### DISCUSSION TITLE
Ente Photos downloads "Large files" using data when "Backup over mobile data" is off

### DISCUSSION THREAD

**Discussion by cyberflamingo** [ente-io/ente#12355] [Enhancements]
### Describe the bug

On Android, when downloading videos below "Large files" through WiFi and then switching to data, Ente Photos continues downloading the videos, even when "Backup over mobile data" is turned off.

### Steps to reproduce

1. Make sure you're on wifi
2. Backup > Backup settings > Backup over mobile data is turned OFF
3. Backup > Free up space > Large files
4. Select some files and start downloading
5. Switch to mobile data

### Expected behavior

When switching to mobile data, the download of any files should stop to prevent data overconsumption.

Ideally, a notification to notice user that phone is not on wifi anymore, and "would you like to resume download anyway?" confirmation.

### Other

Overall, this part of the app is a bit less intuitive. There is no download progress so it's not easy to know if it's working or not, and it seems clicking away from the "Downloading" doesn't actually stop the download (but I may be wrong?)

It's not a core part of the app so not a priority for sure 😄

I will look into the code if I can find the relevant part for the bug above.

### Version

0.9.35

### What product are you using?

Ente Photos

### What platform are you using?

Mobile - Android

**Comment by AmanRajSinghMourya** [comment]
Hey @cyberflamingo, The existing “Backup over mobile data” setting currently applies to automatic backups, not downloads that the user explicitly starts. This separation is also common in other photo and cloud-storage apps.

That said, we agree that a large download continuing after the connection changes from Wi‑Fi to mobile data can result in unexpected data usage. We’ll keep this issue open to track an improved experience—most likely pausing the download and asking whether to continue over mobile data, or providing a separate download-over-mobile-data preference.

Moving this to `Discussions` as this is not an issue but more of a enhancement
